addons/binding/org.openhab.binding.pioneeravr/src/main/java/org/openhab/binding/pioneeravr/protocol/utils/VolumeConverter.java

```diff
@@ -41,9 +44,9 @@ public static double convertFromIpControlVolumeToDb(String ipControlVolume) {
      * @param volumeDb
      * @return the volume for IpControlRequest
      */
-    public static String convertFromDbToIpControlVolume(double volumeDb) {
-        double ipControlVolume = ((MIN_DB_VOLUME + volumeDb) * 2d) + 1d;
-        return formatIpControlVolume(ipControlVolume);
+    public static String convertFromDbToIpControlVolume(double volumeDb, int zone) {
+        double ipControlVolume = ((MIN_DB_VOLUME[zone - 1] + volumeDb) * 2d) + 1d;
```

Comment:
Could you prefix all methods which access the array's in this manner with calling the following method:

```java
private static void validateZone(int zone) {
   if (zone < 1 || zone > 4) {
      throw IllegalArgumentException("An unexpected zone was received, the value should be in the range 1-4");
   }
}
```

In this case we won't have to debug an `ArrayIndexOutOfBoundsException` and its way more clear what went wrong.